framework/post_processors/PostProcessorPrinter_00.cc

```diff
@@ -0,0 +1,205 @@
+#include "PostProcessorPrinter.h"
+#include "event_system/SystemWideEventPublisher.h"
+#include "event_system/EventSubscriber.h"
+#include "event_system/Event.h"
+
+#include "post_processors/PostProcessor.h"
+
+#include "chi_runtime.h"
+#include "chi_log.h"
+
+#include <algorithm>
+
+/**Small utility macro for joining two words.*/
+#define JoinWordsA(x, y) x##y
+/**IDK why this is needed. Seems like counter doesnt work properly without it*/
```

Comment:
@zachhardy @wdarylhawkins @wdhawkins  can you look at this IDK

Reply:
Yes.  Think about what's happening here. You need to invoke JoinWords twice to get the compiler to evaluate __COUNTER__.  So:

JoinWordsB(unique_var_name_ppp_, __COUNTER__) ->
JoinWordsA(unique_var_name_ppp_, <counter_value>) -> 
unique_var_name_ppp_##<counter_value> -> unique_var_name_ppp_<counter_value>

One thing to note about this is that I'm not sure how portable __COUNTER__ is. I seem to recall it's only a GCC (and probably Clang) extension.  Not sure if others have picked it up though.